addons/website_payment/controllers/portal.py

```diff
@@ -132,3 +132,22 @@ def _get_payment_page_template_xmlid(self, **kwargs):
         if kwargs.get('is_donation'):
             return 'website_payment.donation_pay'
         return super()._get_payment_page_template_xmlid(**kwargs)
+
+    @staticmethod
+    def _compute_show_tokenize_input_mapping(providers_sudo, **kwargs):
+        """ Override of payment to hide the "Save my payment details" input in checkout forms
+        when its a donation and user is not logged in.
+
+        :param recordset providers_sudo: The providers for which to determine whether the
+                                         tokenization input should be shown or not, as a sudoed
+                                         `payment.provider` recordset.
```

Comment:
The parent's method's docstring is outdated; use this convention instead. It works for all Odoo extensions:
```suggestion
        :param payment.provider providers_sudo: The providers for which to determine whether the
                                         tokenization input should be shown or not.
```